### PR TITLE
fix: parsing of numbers with parentheses and/or spaces

### LINKED ISSUE #2657: Numbers in parentheses should get converted to -
## Describe the bug
Numbers in parentheses should get converted to - 

This was highlighted in the AI PDF PR. When extracting common financial PDFs there are tons of numbers in parentheses. These should be converted to negative numbers. 

## Expected behavior
Should work like this 
Image

### LINKED ISSUE #2688: Formatting detection for % and currencies
## Describe the bug
Highlighted by PDF imports, it's common to have spaces in between $ and numerical values. In other spreadsheets this gets detected and it's parsed as a number with $ formatting. In Quadratic it's read as a string. 

Image

## Reproduction
Type "$ 50" into a cell, it won't get recognized as #

### PR DESCRIPTION
**Pull request by AyushAgrawal-A2**
## Relevant issue(s)
closes #2657
closes #2688

### CODE REVIEW

**Review by davidfig — APPROVED** [review]

**Review by luke-quadratic — APPROVED** [review]
Works great - to test tried to find many edge cases manually. Then moved on to PDFs and PDFs are being parsing correctly. 

CleanShot 2025-04-10 at 08 33 10@2x

File: quadratic-core/src/controller/operations/cell_value.rs [hunk 1]
```diff
@@ -428,6 +426,150 @@ mod test {
...
+        assert_eq!(
+            format_update.numeric_format,
+            Some(Some(NumericFormat {
+                kind: NumericFormatKind::Currency,
+                symbol: Some("$".to_string()),
+            }))
+        );
+
+        // parentheses with -ve
+        let (value, format_update) = gc.string_to_cell_value("(-$123,456)", true);
```
[the remark this answers is not in the document]
Comment (AyushAgrawal-A2) [reply quadratic-core/src/controller/operations/cell_value.rs:542]: I implemented this as parsing of  `(-$123,456)` will result in `$123,456` but reverted this to just being text to have the sheets / excel
